### ISSUE TITLE
Project graph is broken in 14.8.2 on environments with older git version

### ISSUE THREAD

**Issue by skrtheboss** [nrwl/nx#12338]
## Current Behavior

Project graph is not computed correctly in node:16.17.0 containers, this is caused by the introduction of the `--recurse-submodules` flag in the hasher see https://github.com/nrwl/nx/commit/65e5e148dbdcbbb0dbb3c111e01cef50a2e58583:

## Expected Behavior

Project graph should work on older git versions aswell.

## Steps to Reproduce

```
docker run --rm -it node:16.17.0 /bin/bash
root@97282cc43a86:/# git --version
git version 2.20.1
root@97282cc43a86:/# git ls-files --recurse-submodules -s -z --exclude-standard .
fatal: ls-files --recurse-submodules unsupported mode
```

### Environment

   Node : 16.17.0
   OS   : linux x64
   yarn : 1.22.19
   
   nx : 14.8.2
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.8.2
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 14.8.2
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.8.2
   @nrwl/js : 14.8.2
   @nrwl/linter : 14.8.2
   @nrwl/nest : 14.8.2
   @nrwl/next : Not Found
   @nrwl/node : 14.8.2
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : 14.8.2
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : 14.8.2
   @nrwl/workspace : 14.8.2
   typescript : 4.8.4

**Comment by rpd10** [comment]
Can confirm that downgrading to 14.8.1 fixes the issue for me

**Comment by AgentEnder** [comment]
This should be fixed in 14.8.3

**Comment by jkolyer** [comment]
not fixed in 14.8.3 or 15.0.0

**Comment by AgentEnder** [comment]
@jkolyer that doesn't sound like the issue described here. All of your builds a d tests etc would be failing due to the above as , Nx would be unable to compute hashes.

What is workspace-lint failing with? It doesn't run lint on your projects, it verifies the integrity of your workspace

**Comment by jkolyer** [comment]
> @jkolyer that doesn't sound like the issue described here. All of your builds a d tests etc would be failing due to the above as , Nx would be unable to compute hashes.
> 
> What is workspace-lint failing with? It doesn't run lint on your projects, it verifies the integrity of your workspace

Actually the error I spent a couple hours tracking down was the result of my working around another error with a failing dependency.  So in this case it was not an nx bug.  The nx bug that I spent several more hours dealing with appears to be triggered by the directory nesting level between dependencies.  

To illustrate:
libs/A
libs/B/C

Library C depends on A.  This was not being picked up by automated dependency algorithm.  If I had library B depend on A, it was picked up.  Took many hours of my time tracking this down, including the workaround which brought me here.
